sdk/trace/tracer.go

```diff
@@ -138,7 +138,9 @@ func (tr *tracer) newRecordingSpan(psc, sc trace.SpanContext, name string, sr Sa
 	}
 
 	for _, l := range config.Links() {
-		s.addLink(l)
+		if l.SpanContext.IsValid() {
```

Comment:
would it not be safer to move this condition to the `addLink` method?

Reply:
My reasoning was ignore early and leave the chance of passing zero value SpanContext for any internal purpose. I can change it if you prefer that way.

Reply:
> leave the chance of passing zero value SpanContext for any internal purpose

This is exactly something I was afraid of and why I am proposing this change. Anyway approving - maybe some other reviewer will agree or disagree with my proposal 😉

Reply:
`addLink` should only ever be called from this location (it is incompatible with the specification to add links elsewhere). Splitting the validation logic here and in `addLink`, where it checks the `recordingSpan` is recording and limits are satisfied, would add complexity to the code. I would prefer to add this validation to `addLink` as well, before the lock it taken.